### PR TITLE
Parameterize epsilon for layernorm kernel

### PR DESCRIPTION
**Pull request by kahmed10**
Depends on #1348.

This PR allows for other values of epsilon to be matched when finding layernorm. Similarly, the calculation now uses the variable for epsilon (previously it was a fixed value).

### CODE REVIEW

File: src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp [hunk 1]
```diff
@@ -94,16 +94,17 @@ MIGRAPHX_DPP_REDUCE(op::max, v_max)
...
 {
+    MIGRAPHX_ASSERT(idx.max_nlocal() == idx.nlocal());
 #if __AMDGCN_WAVEFRONT_SIZE == 32
     constexpr index_int lanes_per_thread = 16;
 #else
     constexpr index_int lanes_per_thread = 64;
 #endif
     using type = decltype(f(0));
-    __shared__ type buffer[idx.nlocal() / lanes_per_thread];
+    __shared__ type buffer[idx.max_nlocal() / lanes_per_thread];
```
Comment (TedThemistokleous) [inline src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp:107]: why nlocal to max_nlocal? , if the assert holds above they should be identical no?

File: src/targets/gpu/kernels/include/migraphx/kernels/layernorm.hpp [hunk 2]
```diff
@@ -37,46 +43,45 @@ template <index_int Axis,
...
+        auto variance      = mean_x2 - (mean_x * mean_x);
+        value_type eps_val = eps; // implicit conversion for eps
 
         r.inner([&](auto& y, auto x1, auto x2, auto... xs) {
-            auto m = op(x1, x2) - mean_x;
-            // m * rsqrt(mean(m ^ 2) + 1e-12)
-            y = compute(m * rsqrt(mean_m2 + value_type{1e-12}), xs...);
+            auto x = op(x1, x2);
+            auto m = x - mean_x;
+            // m * rsqrt(mean(m ^ 2) + epsilon)
```
Comment (TedThemistokleous) [inline src/targets/gpu/kernels/include/migraphx/kernels/layernorm.hpp:66]: this intended to be commented out?

**Review by TedThemistokleous — COMMENTED** [review]
Looks good, seems like a lot more moving parts from adding epsilon but looks like you merged in another branch. 

I'll approve once CI goes through.

File: src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp (hunk 1 above)
Comment (kahmed10) [reply src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp:107]: This is not my change, it's #1348

**Review by TedThemistokleous — APPROVED** [review]
Pull in develop lbefore merge. Looks good to me though.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**